engine/runtime-language-epb/src/main/java/org/enso/interpreter/epb/EpbLanguage.java

```diff
@@ -55,11 +32,29 @@ protected void initializeContext(EpbContext context) {
 
   @Override
   protected CallTarget parse(ParsingRequest request) {
-    EpbParser.Result code = EpbParser.parse(request.getSource());
-    ForeignEvalNode foreignEvalNode = ForeignEvalNode.build(this, code, request.getArgumentNames());
+    ForeignEvalNode foreignEvalNode =
+        ForeignEvalNode.build(this, request.getSource(), request.getArgumentNames());
     return foreignEvalNode.getCallTarget();
   }
 
+  static String truffleId(Source langAndCode) {
+    var seq = langAndCode.getCharacters();
+    return seq.subSequence(0, splitAt(seq)).toString().toLowerCase();
+  }
+
+  private static int splitAt(CharSequence seq) {
```

Comment:
I think this has been like that forever but it would be nice if we threw some useful error when there is no `#`

Reply:
Reimplemented in 78d603badb - I like the new structure way more. Thanks for encouraging me to do the cleanup.